File: app/src/main/java/com/doubean/ford/api/GroupSearchResponse.java
package com.doubean.ford.api;

import androidx.annotation.NonNull;

import com.doubean.ford.data.vo.SearchResultItem;

import java.util.ArrayList;
import java.util.List;
/**
 * Data class that represents a group search response from Douban.
 */
public class GroupSearchResponse extends ListResponse<SearchResultItem> {

    private String q;

    public GroupSearchResponse(int start, int count, int total, String q, List<SearchResultItem> items) {
        super(start, count, total, items);
        this.q = q;
    }

    @NonNull
    public List<String> getGroupIds() {
        List<String> groupIds = new ArrayList<>();
        for (SearchResultItem item : getItems()) {
            groupIds.add(item.getGroup().id);
        }
        return groupIds;
    }
}

File: app/src/main/java/com/doubean/ford/api/GroupSearchResponse.kt
package com.doubean.ford.api

import com.doubean.ford.data.vo.SearchResultItem

/**
 * Data class that represents a group search response from Douban.
 */
class GroupSearchResponse(
    start: Int,
    count: Int,
    total: Int,
    private val q: String,
    items: List<SearchResultItem>
) : ListResponse<SearchResultItem>(start, count, total, items) {
    val groupIds: List<String>
        get() {
            val groupIds: MutableList<String> = ArrayList()
            for (item in items) {
                groupIds.add(item.group.id)
            }
            return groupIds
        }
}